Singularity

```diff
@@ -0,0 +1,88 @@
+Bootstrap: library
+From: centos:7.6.1810
+
+%labels
+  # TODO: labels
+
+%help
+  # TODO: help
+
+%environment
+  __conda_setup="$('/usr/local/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
+  if [ $? -eq 0 ]; then
+      eval "$__conda_setup"
+  else
+      if [ -f "/usr/local/etc/profile.d/conda.sh" ]; then
+          . "/usr/local/etc/profile.d/conda.sh"
+      else
+          export PATH="/usr/local/bin:$PATH"
+      fi
+  fi
+  unset __conda_setup
+
+  conda activate base
+
+%post
+  export logpath=/.singularity.d/logs/
+  mkdir -p $logpath
+
+  #  Set default python version if missing
+  PYTHON_VERSION="${PYTHON_VERSION:-3.7.5}"
+  echo "Building for python version: $PYTHON_VERSION"
+
+  #  Basic yum dependencies
+  ##  Install and update
+  yum update -y | tee $logpath/00-yum-update.log
+  yum install -y epel-release | tee $logpath/01-yum-epel-release.log
+  yum groupinstall -y 'Development Tools' | tee $logpath/02-yum-dev-tools.log
+  yum install -y      \
+    nano              \
+    curl              \
+    wget              \
+    tar               \
+    bzip2             \
+    git               \
+    e4fsprogs         \
+    xeyes             \
+    mesa-libGL        \
+    qt5-qtbase        \
+    libxkbcommon-x11  \
+  | tee $logpath/03-yum-install.log
+
+  ##  Log yum packages
+  yum list installed | tee $logpath/04-yum-list-installed.log
+
+  #  Install and setup miniconda
+  ##  Download and install minconda
+  curl -sSL http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o /tmp/miniconda.sh
+  chmod +x /tmp/miniconda.sh
+  /tmp/miniconda.sh -bfp /usr/local/
+  rm -f /tmp/miniconda.sh
+  source /usr/local/bin/activate
+
+  ##  Install python dependencies
+  conda install -y python=$PYTHON_VERSION
+  conda install -y -c anaconda cmake numpy
+  conda install -y -c omgarcia gcc-6
+
+  ##  Download EXtra-foam source
+  mkdir -p /usr/local/src/
+  cd /usr/local/src/
+  git clone https://github.com/European-XFEL/EXtra-foam
+  cd EXtra-foam
+  git checkout $GIT_COMMIT_HASH
+  git submodule update --init | tee -a $logpath/05-extra-foam-submodules.log
+  git log -n 1 | tee -a $logpath/06-extra-foam-clone.log
+
+  export FOAM_WITH_TBB=0
+  export XTENSOR_WITH_TBB=0
+  export FOAM_WITH_XSIMD=0
+  export XTENSOR_WITH_XSIMD=0
```

Comment:
Why do you need those environmental variables here? By default they are all 1, which means to build the parallel version.

Reply:
My bad, it was set like that from when I had some trouble setting up TBB in the image.

Updated the image definition with additional dependencies for TBB and removed those exports, now instead of using conda to install gcc/tbb dependencies it relies on CentOS Software Collections which are a bit more reliable to work with.

Could you check the updated version?

(note: I'm changing the way the install logs are named, so ignore the log naming/counting bit)

Reply:
I see. Indeed there is a bug in my CMake and I am too lazy to fix it: 

The runtime tbb library (`libtbb.so`) can be different from the compile time one. On the online cluster, it always dynamically links to the system `libtbb.so`. Maybe this is what troubled you.

The new version looks good to me. Thanks!

Reply:
The above-mentioned bug has been fixed in #11